src/engraving/dom/cmd.cpp

```diff
@@ -2032,8 +2032,12 @@ static void changeAccidental2(Note* n, int pitch, int tpc)
                 }
             }
         } else {
-            for (Note* nn = n; nn && nn->tieFor(); nn = nn->tieFor()->endNote()) {
-                score->undo(new ChangePitch(nn, pitch, tpc1, tpc2));
+            Note* nn = n;
+            while (nn && nn->tieFor()) {
+                nn = nn->tieFor()->endNote();
+                if (nn) {
+                    score->undo(new ChangePitch(nn, pitch, tpc1, tpc2));
+                }
             }
```

Comment:
Why not:
```suggestion
            for (Note* nn = n; nn; nn = nn->tieFor() ? nn->tieFor()->endNote() : nullptr) {
                score->undo(new ChangePitch(nn, pitch, tpc1, tpc2));
            }
```

Reply:
> Why not:
I think
1. becouse `endNote()` may by `nullptr`, so we would need nullcheck anyway
2. because we dont need to change `n` (which is tie start note), but need to start with next tied note

Reply:
> 1. becouse `endNote()` may by `nullptr`, so we would need nullcheck anyway

That would be done by the `; nn;`. 

Anyway, I'll stop nitpicking about things like this :)